### ISSUE TITLE
SpeechletRequestEnvelope.getContext() returning null

### ISSUE THREAD

**Issue by WillCodeForEver** [amzn/alexa-skills-kit-java#50]
Hello,
SpeechletRequestEnvelope.getContext() returning null. Thus i can't detected supported interfaces to include RenderTemplate & Video DIrectives.

I've double checked that my skill's "Render Template" and "Video" interfaces are enabled in developer.amazon.com

When Testing from "Test" console...i get "NULL" context...

```
	@Override
	public SpeechletResponse onIntent(SpeechletRequestEnvelope<IntentRequest> requestEnvelope) {
```
...

```
	log.debug("====> requestEnvelop: " + requestEnvelope.getClass());
	log.debug("====> request: " + requestEnvelope.getRequest().getClass());
	log.debug("====> context: " + requestEnvelope.getContext());
	if (requestEnvelope.getContext() != null) {
		log.debug("====> context class: " + requestEnvelope.getContext().getClass());
	        log.debug("====> context has display: " + requestEnvelope.getContext().hasState(DisplayInterface.class));
		log.debug("====> context has video: " + requestEnvelope.getContext().hasState(VideoAppInterface.class));
					}
```

I've Skills RenderTemplate 
LOGS:
```
18:43:26
====> requestEnvelop: class com.amazon.speech.json.SpeechletRequestEnvelope

18:43:26
====> request: class com.amazon.speech.speechlet.IntentRequest
18:43:26
====> context: null
```

**Comment by bclement-amazon** [comment]
Hey @WillCodeForEver  

Which test console are you testing from?  Can you send a snippet of the JSON you're sending?

**Comment by WillCodeForEver** [comment]
hey thanks for the reply. I am using Alexa "Test" console under and using the "Text" where I type the utterance and it generates the JSON request:
here is the sample request with some masked attributes
```{
  "session": {
    "sessionId": "SessionId.fxxx",
    "application": {
      "applicationId": "xxx"
    },
    "attributes": {
      "currentState": "LISTING"
    },
    "user": {
      "userId": "xxx"
    },
    "new": false
  },
  "request": {
    "type": "IntentRequest",
    "requestId": "EdwRequestId.24e6c8e1-7b74-408c-bbfd-84214bb73cbc",
    "locale": "en-US",
    "timestamp": "2017-07-03T23:48:57Z",
    "intent": {
      "name": "MyIntent",
      "slots": {
        "Item": {
          "name": "Item",
          "value": "smoothie"
        }
      }
    }
  },
  "version": "1.0"
}```

let me know if you need anything else

**Comment by bclement-amazon** [comment]
Thanks for sending the JSON!

I see why context is null, according to the [Alexa JSON reference](https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/docs/alexa-skills-kit-interface-reference#context-object) "all requests contain the version, context and request objects at the top level". 

The JSON generated by the test console doesn't appear to be populating a context object. Which explains why you're getting a null value.

I've reached out to folks on our side to look into the issue on the test console.  In the interim, you can curate your own context by referencing the [request body syntax](https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/docs/alexa-skills-kit-interface-reference#request-body-syntax) and pasting the contents into the JSON tab on the service simulator.

**Comment by WillCodeForEver** [comment]
thanks for the reply. Yes I did read that doc multiple times and noticed the context object missing from the json. I can use the JSON console but since i am working on a conversational skill where i maintain session variables it will be a lot more effort to copy/paste and adjust the variables for each step.

Thank you for reaching out to the internal team so they can fix the Test Console.

For now I've coded it so if i detect a NULL context i assume the request is from Test Console and it supports the Echo Show directives and return the correct set of directives.
